src/satosa/base.py

```diff
@@ -187,7 +187,7 @@ def _auth_resp_callback_func(self, context, internal_response):
         # If configured construct the user id from attribute values.
         if "user_id_from_attrs" in self.config["INTERNAL_ATTRIBUTES"]:
             subject_id = [
-                "".join(internal_response.attributes[attr]) for attr in
+                "".join(internal_response.attributes.get(attr, '')) for attr in
```

Comment:
There are multiple things that can go wrong here. 

- If we use `.get` then `None` is a possible value, which will end up being concatenated as a string.
- Some attributes have multiple values; `attributes[attr]` could itself be a list of values.

Reply:
I'm using this in production, of and attr Is unavailable It Will have ''. You told me about a strategy to get It to work without this patch, Just put here an example, I Will test, then we could close this PR if you agree